### ISSUE TITLE
Map ignores retainedFileCountLimit

### ISSUE THREAD

**Issue by markat1** [serilog/serilog-sinks-map#29]
Hi,
I have a problem regarding limiting the number of log files.

I have retainedFileCountLimit  2, but map seems to ignore this rule. Unfortunately it creates more than 2 files.

```
 .WriteTo.Async(a =>
            {
                a.Map<string>(
                    e => DateTime.Now.ToString("dd_MM_yyyy"),
                    (v, wt) => wt.File($"{AppDomain.CurrentDomain.BaseDirectory}/logs/log_{v}.txt",
                                       retainedFileCountLimit: 2,
                                       outputTemplate: "[{Timestamp:dd/MM/yyyy HH:mm:ss} {Level:u3}] {Message:lj}{NewLine}{Exception}",
                                       buffered: true,
                                       fileSizeLimitBytes: 105000000),
                    sinkMapCountLimit: 1);
            }, bufferSize: 120000)
```

it should only have been 2 files, but I see 3 files:

```
log_15_05_2021.txt
log_16_05_2021.txt
log_17_05_2021.txt
```

**Comment by overlego** [comment]
getting the same issue, anyone got a fix?

**Comment by bartelink** [comment]
The Map Sink does not implement limiting/trimming - it just fans out over an unlimited number of sink configurations

The File Sink does implement trimming, but [does not let you customize the naming strategy](https://github.com/serilog/serilog-sinks-file/issues/243)
It's way beyond the remit of Map to manage sinks over time (and Sinks would need to impleent some generic way to route requests to trim if it decided there were too many etc)

The bottom line here is that
- The Map Sink does not (and can not) implement a `retainedFileCountLimit`
- the File Sink implements a `retainedFileCountLimit`

Thus the most likely way to achieve your goal is to:
- use the File sink (but that does imply accepting [the limitations of its naming strategies](https://github.com/serilog/serilog-sinks-file/issues/243))
- possibly use the File's Sink's Lifecycle hooks to customise things

**Comment by bartelink** [comment]
Duplicate of #15 (feature request to implement generic trimming)
